src/app/dao/tests/JoinerDAO.spec.ts

```diff
@@ -1,17 +1,438 @@
+            // When the candidate accepts the config by setting partStatus
+            const result: Promise<void> = joinerDAO.update(partId, { partStatus: PartStatus.PART_STARTED.value });
+
+            // Then it should succeed
+            await expectAsync(result).toBeResolvedTo();
+        });
+        it('should forbid chosen opponent to change status to any other value than STARTED', async() => {
+            const partStatusValues: number[] = [
+                PartStatus.PART_CREATED.value,
+                PartStatus.PART_FINISHED.value,
+            ];
+
+            for (const partStatus of partStatusValues) {
+                // When the candidate tries to set part status to any other value
+                const result: Promise<void> = joinerDAO.update(partId, { partStatus });
+
+                // Then it should fail
+                await expectFirebasePermissionDenied(result);
+            }
+        });
+        it('should forbid chosen opponent to change status if config is not proposed', async() => {
+            // And given that the part is not proposed
+            await reconnectUser('foo@bar.com');
+            const update: Partial<Joiner> = { partStatus: PartStatus.PART_CREATED.value };
+            await expectAsync(joinerDAO.update(partId, update)).toBeResolvedTo();
+            await signOut();
+
+            await reconnectUser('bar@bar.com');
+
+            const partStatusValues: number[] = [
+                PartStatus.CONFIG_PROPOSED.value,
+                PartStatus.PART_STARTED.value,
+                PartStatus.PART_FINISHED.value,
+            ];
+
+            for (const partStatus of partStatusValues) {
+                // When the candidate tries to set part status to any other value
+                const result: Promise<void> = joinerDAO.update(partId, { partStatus });
+
+                // Then it should fail
+                await expectFirebasePermissionDenied(result);
+            }
+        });
+        it('should forbid non-chosen candidate to change status to any value', async() => {
+            // And given any other user who is not candidate
+            await createConnectedGoogleUser('baz@bar.com', 'not-candidate');
+
+            const partStatusValues: number[] = [
+                PartStatus.PART_CREATED.value,
+                PartStatus.PART_STARTED.value,
+                PartStatus.PART_FINISHED.value,
+            ];
+
+            // When trying to change the status to any value
+            for (const partStatus of partStatusValues) {
+                // When the candidate tries to set part status to any other value
+                const result: Promise<void> = joinerDAO.update(partId, { partStatus });
+
+                // Then it should fail
+                await expectFirebasePermissionDenied(result);
+            }
+        });
+    });
+    it('should allow verified users to read the joiner', async() => {
```

Comment:
pour la beauté du pattern, du wrapperais ça dans un describe('verified user that are not candidate yet') ?

Reply:
fait en role-based yep!